pkg/controllers/addons_controller.go

```diff
@@ -161,12 +153,12 @@ func (r *AddonsLayerReconciler) processAddonLayer(l layers.Layer) error {
 	}
 
 	err := r.processPrune(l)
-	if err != nil {
+	if err != nil || l.IsDelayed() {
```

Comment:
This doesn't make sense.  Why would you want the processAddonLayer function to return an error when there isn't an error?

If the IsUpdated flag shouldn't be used for this purpose, I'll add a return value to indicate that processing is done and skip the processApply and checkSuccess calls if that return value is set.  It adds a bit of complexity to the function, but avoids the need for any additional flags, and seems clearer overall.